infra/azure/__main__.py

```diff
@@ -398,3 +406,252 @@ def get_kubeconfig(
         depends_on=[minio_tenant],
     ),
 )
+
+## Argo Workflows
+argo_server_ns = Namespace(
+    "argo-server-ns",
+    metadata=ObjectMetaArgs(
+        name="argo-server",
+    ),
+    opts=ResourceOptions(
+        provider=k8s_provider,
+        depends_on=[managed_cluster],
+    ),
+)
+
+argo_workflows_ns = Namespace(
+    "argo-workflows-ns",
+    metadata=ObjectMetaArgs(
+        name="argo-workflows",
+    ),
+    opts=ResourceOptions(
+        provider=k8s_provider,
+        depends_on=[managed_cluster],
+    ),
+)
+
+argo_url = Output.format(
+    "{0}.{1}",
+    config.require("argo_url_prefix"),
+    config.require("base_fqdn"),
+)
+
+argo_sso_secret = Secret(
+    "argo-server-sso-secret",
+    metadata=ObjectMetaArgs(
+        name="argo-server-sso",
+        namespace=argo_server_ns.metadata.name,
+    ),
+    type="Opaque",
+    string_data={
+        "client-id": config.require_secret("entra_app_client_id"),
+        "client-secret": config.require_secret("entra_app_client_secret"),
+    },
+    opts=ResourceOptions(
+        provider=k8s_provider,
+        depends_on=[argo_server_ns],
+    ),
+)
+
+argo_minio_secret = Secret(
+    "argo-minio-secret",
+    metadata=ObjectMetaArgs(
+        name="argo-artifacts-minio",
+        namespace=argo_server_ns.metadata.name,
+    ),
+    type="Opaque",
+    string_data={
+        "accessKey": config.require_secret("minio_root_user"),
+        "secretKey": config.require_secret("minio_root_password"),
+    },
+    opts=ResourceOptions(
+        provider=k8s_provider,
+        depends_on=[argo_server_ns],
+    ),
+)
+
+argo_workflows = Chart(
+    "argo-workflows",
+    namespace=argo_server_ns.metadata.name,
+    chart="argo-workflows",
+    version="0.45.12",
+    repository_opts=RepositoryOptsArgs(
+        repo="https://argoproj.github.io/argo-helm",
+    ),
+    value_yaml_files=[
+        FileAsset("./k8s/argo_workflows/values.yaml"),
+    ],
+    values={
+        "controller": {"workflowNamespaces": [argo_workflows_ns.metadata.name]},
+        "server": {
+            "ingress": {
+                "hosts": [argo_url],
+                "tls": [
+                    {
+                        "secretName": "argo-ingress-tls-letsencrypt",
+                        "hosts": [argo_url],
+                    }
+                ],
+            },
+            "sso": {
+                "enabled": True,
+                "issuer": Output.concat(
+                    "https://login.microsoftonline.com/",
+                    config.require_secret("entra_tenant_id"),
+                    "/v2.0",
+                ),
```

Comment:
Maybe this would be better as a variable like `sso_issuer_url`.

If we don't want to make users put the full url, it could be something like

```
match oidc_provider:
    case "entra":
        sso_issuer_url = f"https://login.microsoftonline.com/{entra_tenant_id}..."
    case "...":
        ...
```